### PR TITLE
desktop: Add theme preference

### PR DESCRIPTION
**Pull request by kjarosh**
Related to https://github.com/ruffle-rs/ruffle/issues/17288.

This PR adds a setting for GUI theme. The default value is `System`, which matches the theme with system's theme and watches for theme changes.

The theme-related code became somewhat complicated, so I added `ThemeController` to make the code more readable.

### CODE REVIEW

File: desktop/src/gui/theme.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        this.start_dbus_theme_watcher_linux().await;
+        this.start_theme_preference_watcher(&preferences).await;
+
+        this.set_theme_preference(preferences.theme_preference())
+            .await;
+
+        this
+    }
+
+    async fn start_theme_preference_watcher(&self, preferences: &GlobalPreferences) {
```
Comment (sleepycatcoding) [inline desktop/src/gui/theme.rs:52]: This does not need to be async?

Comment (kjarosh) [reply desktop/src/gui/theme.rs]: Right, fixed!

File: desktop/src/gui/theme.rs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            }
+            ThemePreference::Light => Theme::Light,
+            ThemePreference::Dark => Theme::Dark,
+        };
+        self.set_theme_internal(data, theme);
+    }
+
+    #[cfg(target_os = "linux")]
+    async fn get_system_theme(&self) -> Result<Theme, Box<dyn Error>> {
+        let connection = zbus::Connection::session().await?;
```
Comment (sleepycatcoding) [inline desktop/src/gui/theme.rs:151]: Can you not reuse the open connection in self?

Comment (kjarosh) [reply desktop/src/gui/theme.rs]: Whoops, of course, that was the main reason of the refactor, just... forgot about that :smile: thanks!

**Comment by torokati44** [comment]
Oh, look: https://github.com/rust-windowing/winit/issues/3837, https://github.com/rust-windowing/winit/issues/3838

**Comment by Dinnerbone** [comment]
Oh wow, light mode is so ugly.
Thank you, I guess.

oh no.

**Review by Dinnerbone — APPROVED** [review]

**Comment by evilpie** [comment]
Thank you!

**Pull request merged**